### PR TITLE
Added _T{n} suffix to all samples for check_samplesheet to work

### PR DESCRIPTION
**Pull request by praveenraj2018**
To address the issue #1632 

Modified bin/check_samplesheet.py to add `_T{n}` suffix to all samples for the merging samples at downstream inside the pipelines to work properly.

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/pipeline-template/bin/check_samplesheet.py [hunk 1]
```diff
@@ -113,19 +113,16 @@ def validate_unique_samples(self):
         """
         Assert that the combination of sample name and FASTQ filename is unique.
 
-        In addition to the validation, also rename the sample if more than one sample,
-        FASTQ file combination exists.
+        In addition to the validation, also rename all samples to have a suffix of _T{n}, where n is the
+        number of times the same sample exist, but with different FASTQ file(s).
```
Comment (Midnighter) [inline nf_core/pipeline-template/bin/check_samplesheet.py:117]:
```suggestion
        number of times the same sample exist, but with different FASTQ files, e.g., multiple runs per experiment.
```

**Review by Midnighter — COMMENTED** [review]
In essence, this part of the script was already supposed to do what you describe but only when there are more than one run per sample. Is that not doing what you would expect?

File: nf_core/pipeline-template/bin/check_samplesheet.py (hunk 1 above)
Comment (drpatelh) [reply nf_core/pipeline-template/bin/check_samplesheet.py:117]: No. The idea is to always add a `T*` suffix even if there is one sample. That way it allows us to assume everything after the last "_" needs to be trimmed off [here](https://github.com/nf-core/rnaseq/blob/89bf536ce4faa98b4d50a8ec0a0343780bc62e0a/workflows/rnaseq.nf#L199)

**Review by drpatelh — APPROVED** [review]
LGTM!

**Pull request merged**